core/arch/arm/include/kernel/linker.h

```diff
@@ -22,33 +22,59 @@ extern const uint8_t __exidx_end[];
 extern const uint8_t __extab_start[];
 extern const uint8_t __extab_end[];
 
+/*
+ * Warning! (ASLR) Do not use any of the _SZ_С defines or _size[] variables
+ * below in code that runs after dynamic relocations have been applied. Their
+ * value could be *wrong* after relocation (for instance Aarch64 GCC 9.2
+ * generates dynamic relocations for the _size[] symbols, but Clang 11.0.0 does
+ * not). Instead, use the defines with _SZ_R suffix. The _SZ_C macros are
+ * mostly here for use in register_phys_mem() where the _SZ_R macros would be
+ * rejected by the compiler ('initializer element is not constant'). Other uses
+ * are likely to be incorrect.
+ * Reiterating, _SZ_C defines are for compile-time definitions, when constant
+ * values are required and _SC_R are for run-time uses.
+ */
+
 #define VCORE_UNPG_RX_PA	((unsigned long)__vcore_unpg_rx_start)
-#define VCORE_UNPG_RX_SZ	((size_t)__vcore_unpg_rx_size)
+#define VCORE_UNPG_RX_SZ_C	((size_t)__vcore_unpg_rx_size)
```

Comment:
To discourage the use of the `_C` defines perhaps they should be having a `_UNSAFE` suffix instead.
We could then drop the `_R` suffix.